nilearn/plotting/img_plotting.py

```diff
@@ -36,6 +36,44 @@
 ################################################################################
 # Core, usage-agnostic functions
 
+def _get_plot_stat_map_params(stat_map_img, vmax, symmetric_cbar, kwargs, force_min_stat_map_value=None):
+# make sure that the color range is symmetrical
+    if vmax is None or symmetric_cbar in ['auto', False]:
+        stat_map_img = _utils.check_niimg_3d(stat_map_img)
+        stat_map_data = stat_map_img.get_data()
+    # Avoid dealing with masked_array:
+        if hasattr(stat_map_data, '_mask'):
+            stat_map_data = np.asarray(stat_map_data[np.logical_not(stat_map_data._mask)])
+        stat_map_max = np.nanmax(stat_map_data)
+        if force_min_stat_map_value == None:
+            stat_map_min = np.nanmin(stat_map_data)
+        else:
+            stat_map_min = force_min_stat_map_value
+    if symmetric_cbar == 'auto':
+        symmetric_cbar = stat_map_min < 0 and stat_map_max > 0
+    if vmax is None:
+        vmax = max(-stat_map_min, stat_map_max)
+    if 'vmin' in kwargs:
+        raise ValueError('this function does not accept a "vmin" '
+            'argument, as it uses a symmetrical range '
+            'defined via the vmax argument. To threshold '
+            'the map, use the "threshold" argument')
+    vmin = -vmax
+    if not symmetric_cbar:
+        negative_range = stat_map_max <= 0
+        positive_range = stat_map_min >= 0
+        if positive_range:
+            cbar_vmin = 0
+            cbar_vmax = None
+        elif negative_range:
+            cbar_vmax = 0
+            cbar_vmin = None
+        else:
+            cbar_vmin = stat_map_min
+            cbar_vmax = stat_map_max
+    else:
+        cbar_vmin, cbar_vmax = None, None
+    return cbar_vmin, cbar_vmax, stat_map_img, vmin, vmax
```

Comment:
I guess I missed the fact that this function does the check_niimg_3d.

I would personally be in favor of doing the check_niimg_3d in plot_glass_brain and plot_stat_map as this is quite common to do that for nilearn functions that take Niimg-like object.